### PR TITLE
feat(x/data): unique constraints on resolver url and manager

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

Closes: #1212

This pull request adds unique constraints on resolver url and manager combination.
 
---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/regen/data/v1/state.proto [hunk 1]
```diff
@@ -62,6 +62,7 @@ message Resolver {
     primary_key : {fields : "id", auto_increment : true}
     index : {id : 1, fields : "url"}
     index : {id : 2, fields : "manager"}
```
Comment (technicallyty) [inline proto/regen/data/v1/state.proto:64]: was wondering if the new index below allows us to remove any of these other ones, and i couldn't seem to find any with the `manager` index being used. is it still needed?

**Review by technicallyty — APPROVED** [review]
lgtm

File: proto/regen/data/v1/state.proto (hunk 1 above)
Comment (ryanchristo) [reply proto/regen/data/v1/state.proto]: Looks like it wasn't being used. Removed in https://github.com/regen-network/regen-ledger/pull/1219/commits/73aeb6ea45818c7b2f30d7b85ba6eac6bc873e4a.

**Review by aleem1314 — APPROVED** [review]
lgtm

**Pull request merged**